staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -820,7 +820,14 @@ func (b *Builder) visitorResult() *Result {
 	}
 
 	if len(b.resources) != 0 {
-		return &Result{err: fmt.Errorf("resource(s) were provided, but no name, label selector, or --all flag specified")}
+		for _, r := range b.resources {
+			_, err := b.mappingFor(r)
+			if err != nil {
+				return &Result{err: err}
+			} else {
+				return &Result{err: fmt.Errorf("resource(s) were provided, but no name, label selector, or --all flag specified")}
```

Comment:
It seems this return should be placed outside the for loop.

Reply:
Actually i think this will just return the error of first element. So we don't have to check the others.

Reply:
I meant that we should continue calling mappingFor() for the rest of b.resources.
'but no name' error should only be emitted when we have traversed b.resources

If you agree, I can send a PR.

Reply:
Thanks, I think maybe it's just need to do like `b.mappingFor(resources[0])`. Or maybe you are right about check all others. I'm not sure sorry.

Reply:
the point is that resources[1] maybe the resource which you want to see the newly added message.
Checking resources[0] alone doesn't serve your original purpose.